src/App/integrationTests/App.dontshowOfflineDeleted.test.js

```diff
@@ -18,13 +23,22 @@ test('Collect page only shows records that arent marked to be deleted next sync'
     _deleted: true,
   })
 
-  renderAuthenticatedOffline(<App dexieCurrentUserInstance={dexieCurrentUserInstance} />, {
-    initialEntries: ['/projects/5/collecting/'],
-    dexiePerUserDataInstance,
-    dexieCurrentUserInstance,
-  })
+  const { user } = renderAuthenticatedOffline(
+    <App dexieCurrentUserInstance={dexieCurrentUserInstance} />,
+    {
+      initialEntries: ['/projects/5/collecting/'],
+      dexiePerUserDataInstance,
+      dexieCurrentUserInstance,
+    },
+  )
+
+  const pageSizeSelector = await screen.findByTestId('page-size-selector')
 
-  userEvent.selectOptions(await screen.findByTestId('page-size-selector'), '20')
+  await waitFor(() => within(pageSizeSelector).getByText('20'))
```

Comment:
Example of something needing to be converted to use `waitFor`

Reply:
would we have to do this for every async test?

Reply:
No, its just something to pull out of your toolbox if your test isnt behaving like it should. This is actually wrong technically, but coincidentally worked. I forgot to wrap the waitfor innards in an expect. I changed it to `await waitFor(() => expect(within(pageSizeSelector).getByText('20')))`

Reply:
ah ok - thanks for clarifying